share/functions/__fish_print_portage_repository_paths.fish

```diff
@@ -1,4 +1,14 @@
 function __fish_print_portage_repository_paths --description 'Print the paths of all configured repositories'
-    # repos.conf may be a file or a directory
-    find /etc/portage/repos.conf -type f -exec cat '{}' + | string replace -r --filter '^\s*location\s*=\s*(\S+)' '$1'
+    set -l a /etc/portage/repos.conf
+    set -l b
+    set -l c /usr/share/portage/config/repos.conf
+    test -d $a
+    and set b (find $a -type f )
+    test -f $a
+    and set b $a
+    test -n (echo $b)
+    and string match -q "[gentoo]" (cat $b)
```

Comment:
there's a subtle problem here: what if $b contains a line starting with `--foo`.
`string match` will try to parse that as option and fail.

there are multiple ways to avoid this:

```suggestion
    and string match -q "[gentoo]" -- (cat $b)
```

or

```suggestion
    and cat $b | string match -q "[gentoo]"
```

or

```suggestion
    grep -Fxq "[gentoo]" $b
```